daemons/attrd/attrd_utils.c

```diff
@@ -103,6 +103,10 @@ attrd_shutdown(int nsig)
         peer_protocol_vers = NULL;
     }
 
+    attrd_unregister_handlers();
+    pcmk_cluster_disconnect(attrd_cluster);
```

Comment:
I'm thinking this should be just before `pcmk_cluster_free()` (really that function should probably do it ...)